src/Indexer.ts

```diff
@@ -0,0 +1,73 @@
+import * as path from 'path'
+import * as url from 'url'
+
+import * as ts from 'typescript'
+
+import { Input } from './Input'
+import * as lsif from './lsif'
+import { LsifSymbol } from './LsifSymbol'
+import { Options, lsif_typed } from './main'
+import { Packages } from './Packages'
+import { Visitor } from './Visitor'
+
+export class Indexer {
+  public options: Options
+  public program: ts.Program
+  public checker: ts.TypeChecker
+  public symbolsCache: Map<ts.Node, LsifSymbol> = new Map()
```

Comment:
Nit: cache already conveys a collection, so I think we should name this `symbolCache` instead.